### PR TITLE
OSL-386: Execute data deletion from /deleteUserData express endpoint

### PR DESCRIPTION
**Pull request by katerinachinnappan**
## Goal

Legal said we don't need to retain any data. When a user deletes their account, `/deleteUserData` endpoint will delete all shareable list data from the db for that user. Added integration test for `/deleteUserData` endpoint.

## Tickets

- [https://getpocket.atlassian.net/browse/OSL-386](https://getpocket.atlassian.net/browse/OSL-386)

### CODE REVIEW

File: src/public/routes/deleteUserData.ts [hunk 1]
```diff
@@ -36,6 +40,76 @@ export function validate(req: Request, res: Response, next: NextFunction) {
...
+      userId,
+    },
+    orderBy: { updatedAt: 'desc' },
+    include: {
+      listItems: true,
+    },
+  });
+  const shareableListIds = [];
+  shareableLists.forEach(function (list) {
+    shareableListIds.push(parseInt(list.id as unknown as string));
```
Comment (jpetto) [inline src/public/routes/deleteUserData.ts:61]: nit - could we put a comment on this line as to why we have to double cast?

File: src/public/routes/deleteUserData.ts [hunk 2]
```diff
@@ -36,6 +40,76 @@ export function validate(req: Request, res: Response, next: NextFunction) {
...
+async function deleteShareableListUserData(
+  userId: number | bigint,
+  requestId: string
+): Promise<string> {
+  const shareableListIds = await getAllShareableListIdsForUser(userId);
+  // if there are no lists found for a userId, lets not make unecessary calls
+  // to the db
+  if (shareableListIds.length === 0) {
+    return `No shareable list data to delete for User ID: ${userId} (requestId='${requestId}')`;
+  } else if (shareableListIds.length > 0) {
```
Comment (jpetto) [inline src/public/routes/deleteUserData.ts:94]: nit - you could drop the `else` here, as when the `if` is truthy, the function returns.

**Review by jpetto — CHANGES_REQUESTED** [review]
the approach is good - just some comments and suggestions around tightening things up.

File: src/public/routes/deleteUserData.ts [hunk 3]
```diff
@@ -44,13 +118,23 @@ router.post(
     const requestId = req.body.traceId ?? nanoid();
```
[the remark this answers is not in the document]
Comment (katerinachinnappan) [reply src/public/routes/deleteUserData.ts:118]: @jpetto I am actually not sure, this seems to be the pattern in other repos and I have assumed this is part of the POST request lambda sends to the endpoint, to trace the request.

[the remark this answers is not in the document]
Comment (jpetto) [reply src/public/routes/deleteUserData.ts:112]: hmm, okay. after some looking, i _think_ this pattern came from kat with the intent of tracing a request through the apollo gateway down to the specific subgraph (to match errors in the gateway to errors in the subgraph). as in this case we are going directly from a lambda to this service, i don't think we need to add this `requestId`. if that sounds right to you, i say we remove it to keep the code clean.

but - let me know if you do see a use case!

**Review by jpetto — CHANGES_REQUESTED** [review]
thanks for addressing all the feedback! looks great. one last thing to remove the `requestId`, then i think we'll be good.

File: src/public/routes/deleteUserData.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (katerinachinnappan) [reply src/public/routes/deleteUserData.ts]: @jpetto if this is the case, then lets discard it!

**Review by jpetto — APPROVED** [review]
nice work! thanks for all the changes on this one!

**Pull request merged**